test/tools/runner/filters/client_encryption_filter.js

```diff
@@ -43,10 +43,23 @@ class ClientSideEncryptionFilter {
       test.metadata && test.metadata.requires && test.metadata.requires.clientSideEncryption;
 
     // CSFLE is only supported on LTS versions of node
-    const nodeSupportsCSFLE = semver.satisfies(process.version, '>4');
+    // the node filter will need to run after this one
+    if (clientSideEncryption) {
+      if (!test.metadata) {
+        test.metadata = {};
+      }
+      if (!test.metadata.requires) {
+        test.metadata.requires = {};
+      }
+      if (!test.metadata.requires.node) {
+        test.metadata.requires.node = '>4';
+      } else if (!semver.satisfies(process.version, '>4')) {
+        test.metadata.requires.node = '>4';
+      }
+    }
```

Comment:
A filter should not mutate the code it's observing. I suspect what's really wrong here has something to do with `ret` below, but the goal of this filter is that IFF `clientSideEncryption` is attached to a test or test suite then it should apply these filtering rules, otherwise the filter should be skipped.

Reply:
I completely agree, ran to the fast solution before the best one here. 
It will now return true for all tests that don't specify `clientSideEncryption` and check the node version if `clientSideEncryption` if defined.